docs/configuration_options.md

```diff
@@ -0,0 +1,303 @@
+# Configuration Reference
+
+TODO: SECTIONS(DELAYED JOB SUPPORT, ASYNC SUPPORT, WRITE TO FILE, PERSON TRACKING)
+
+<!-- Sub:[TOC] -->
+
+## access_token
+
+**Required**
+
+Sets the access token used to send payloads to rollbar.
+
+Items sent through a given access token arrive in that access token's project
+and respect the rate limits set on that access token.
+
+## async_handler
+
+If `config.use_async = true` explicitly sets the function used to send
+asynchronous payloads to Rollbar. Not needed if using one of the built in async
+reporters:
+
+ * girl_friday
+ * sucker_punch
+ * Sidekiq
+ * Resque
+ * threading
+
+## branch
+
+**Default** "master"
+
+Name of the checked-out source control branch.
+
+## code_version
+
+A string, up to 40 characters, describing the version of the application code.
+
+Rollbar understands these formats:
+
+ * semantic version (i.e. "2.1.12")
+ * integer (i.e. "45")
+ * git SHA (i.e. "3da541559918a808c2402bba5012f6c60b27661c")
+
+## custom_data_method
+
+TODO: What does it do? Demo?
```

Comment:
@Crisfole deep_copy call is there in order to be sure we don't mutate never the value received from the custom_data_method.

So, we call `configuration.custom_data_method.call` and then we do a deep copy with the result, so it's a new object and can't the result object.